src/mqt/predictor/rl/predictorenv.py

```
@@ -623,34 +629,84 @@ def determine_valid_actions_for_state(self) -> list[int]:
+                actions.extend(self.actions_routing_indices)
+                actions.extend(self.actions_structure_preserving_indices)
+            elif not synthesized and laid_out and routed:
+                actions.extend(self.actions_synthesis_indices)
+                actions.extend(self.actions_structure_preserving_indices)
+            else:
+                actions.append(self.action_terminate_index)
+                actions.extend(self.actions_structure_preserving_indices)
+                actions.extend(self.actions_final_optimization_indices)
+
+        else:  # flexible
+            if not synthesized and not laid_out:
```

Ruff (0.16.1)

[warning] Use `elif` instead of `else` then `if`, to reduce indentation

Convert to `elif`

(PLR5501)